eth/eventhandler/handlers.go

```diff
@@ -435,7 +422,7 @@ func (eh *EventHandler) handleClusterReactivated(txn basedb.Txn, event *contract
 
 	// bump slashing protection for operator reactivated validators
 	for _, share := range toReactivate {
-		if err := eh.keyManager.(ekm.StorageProvider).BumpSlashingProtection(share.SharePubKey); err != nil {
+		if err := eh.keyManager.BumpSlashingProtection(phase0.BLSPubKey(share.SharePubKey)); err != nil {
```

Comment:
I guess this is why we need the SlashingProtector struct?

Reply:
We need the `SlashingProtector` struct to extract slashing protection logic from the local signer and apply it to both local and remote without copying the code. It used to be a part of the local signer (the original ekm code) but I had to embed the local signer into the remote signer to use the slashing protection code. 

This particular change is just an improvement because interface type assertions may be dangerous